File: JS 문제 풀이/폰켓몬.js
function solution(nums) {
    let answer = 0;
    let set = new Set(nums);
    let size = nums.length/2;

    for(let num of nums) set.add(num);

    if(set.size>size) answer=size;
    else answer=set.size;

    return answer;
}

/*
최대한 많은 종류의 폰켓몬을 가져가는 경우에서 폰켓몬의 종류 갯수
Tip. set의 사이즈는 size이다.

*/

File: JAVA 문제풀이/프로그래머스 Lv.1/폰켓몬.java
import java.util.HashSet;

class Solution {
    public int solution(int[] nums) {
        int answer = 0;
        /*
            폰켓몬 n마리중 n/2 마리를 가져갈 수있다.
            각 폰켓몬은 번호가 주어짐
            가질수 있는 폰켓몬의 종류의 수
        */
        HashSet<Integer> set = new HashSet<Integer>();

        for(int i=0;i<nums.length;i++){
            set.add(nums[i]);
        }

        if(nums.length/2>=set.size()){
            answer=set.size();
        }else{
            answer=nums.length/2;
        }
        System.out.println(set.size());

        return answer;
    }
}
